### PR TITLE
Banner Collapse TM-2374

### PR DESCRIPTION
**Pull request by mjoyce91**

### CODE REVIEW

**Comment by SidelineCory24** [comment]
Left to do:
- [ ] Put the button in the middle
- [ ] Update styling for line break line
- [ ] Need to test when there is no text at all/general error handling

**Comment by SidelineCory24** [comment]
Did we decide if we wanted the pencil at the end of the message or not?
image

image

**Comment by mjoyce91** [comment]
> Did we decide if we wanted the pencil at the end of the message or not? image
> 
> image

I think it's fine there since only admins will see it

File: src/utilities.test.js [hunk 1]
```diff
@@ -906,3 +907,22 @@ describe('move', () => {
     expect(exp).toEqual([2, 1, 3]);
   });
 });
+
+describe('splitByLineBreak', () => {
```
Comment (SidelineCory24) [inline src/utilities.test.js:911]: 😎

**Comment by SidelineCory24** [comment]
@elizabeth-jimenez @scott062 I did PP with Mike for this ticket, so I'll leave it to one of you to approve. Everything looks good on my end though 🚀

File: src/Components/Header/BetaHeader/BetaHeader.test.jsx [hunk 2]
```diff
@@ -54,6 +54,62 @@ describe('BetaHeader', () => {
...
+
+  it('displays the correct button text when state.headerDropdownVisible is changed', () => {
+    const data = 'test\n\n\ntest';
+    const wrapper = shallow(
+      <BetaHeader.WrappedComponent data={data} />,
+    );
+    expect(wrapper.find('button').text()).toBe('2Show alerts <FontAwesome />');
+    wrapper.instance().setState({ headerDropdownVisible: true });
+    wrapper.update();
+    expect(wrapper.find('button').text()).toBe('2Hide alerts <FontAwesome />');
```
Comment (elizabeth-jimenez) [inline src/Components/Header/BetaHeader/BetaHeader.test.jsx:100]: 🙌

File: src/sass/_betaHeader.scss [hunk 3]
```diff
@@ -32,5 +33,69 @@
...
+    position: relative;
+  }
+
+  .banner-toggle-button {
+    margin: 0;
+    position: absolute;
+    bottom: -19px;
+    z-index: 1000;
+    border-radius: 20px;
+    background-color: #fdf0cc;
```
Comment (elizabeth-jimenez) [inline src/sass/_betaHeader.scss:67]: should we make all these hex colors variables?

File: src/utilities.test.js [hunk 4]
```diff
@@ -906,3 +907,22 @@ describe('move', () => {
...
+  it('splits no messages', () => {
+    expect(splitByLineBreak(null)).toEqual(['']);
+  });
+
+  it('splits 2 messages', () => {
+    expect(splitByLineBreak('test\n\n\ntest2')).toEqual(['test', 'test2']);
+  });
+
+  it('splits 5 messages', () => {
+    expect(splitByLineBreak('test \n a  \n\n\ntest2 a\n\n\n test 3 \n\n\n test4 4 4 \n\n\ntest5 \n\n').length).toEqual(5);
```
Comment (elizabeth-jimenez) [inline src/utilities.test.js:926]: ~How come this isn't 6? 😱 ~ nvm i see why!

File: src/utilities.js [hunk 5]
```diff
@@ -879,3 +879,5 @@ export const useCloseSwalOnUnmount = () =>
     } catch { return null; }
     return null;
   }, []);
+
+export const splitByLineBreak = text => (text || '').split('\n\n\n');
```
Comment (elizabeth-jimenez) [inline src/utilities.js:883]: Should we filter out the empties after split?
"message /n/n/n " is valid as 2

File: src/sass/_betaHeader.scss [hunk 6]
```diff
@@ -1,13 +1,14 @@
...
     margin-right: 1em;
   }
 
   .loader {
-    display: inline;
+    display: inline-block;
+    width: 100%;
 
     > div {
       display: inline;
```
Comment (elizabeth-jimenez) [inline src/sass/_betaHeader.scss:14]: Now that they can add more than one message, should we give them more editing height on line 25 (we could even take some from the margin-bottom)?

**Comment by elizabeth-jimenez** [comment]
This is a really cool feature Cory and Mike 🦜

**Review by elizabeth-jimenez — COMMENTED** [review]
Just a few comments

File: src/sass/_betaHeader.scss (hunk 3 above)
Comment (mjoyce91) [reply src/sass/_betaHeader.scss:67]: ah yes we should 👍

File: src/utilities.js (hunk 5 above)
Comment (mjoyce91) [reply src/utilities.js:883]: My thought was that if they were included, it would have been intentional

File: src/sass/_betaHeader.scss (hunk 6 above)
Comment (mjoyce91) [reply src/sass/_betaHeader.scss:14]: That's a good idea. It was definitely annoying to edit with how short it was

File: src/utilities.js (hunk 5 above)
Comment (elizabeth-jimenez) [reply src/utilities.js:883]: ah, hadn't thought about it being intentional

**Comment by elizabeth-jimenez** [comment]
image
1280*800 with new height

**Review by elizabeth-jimenez — APPROVED** [review]
👍

**Pull request merged**